09b_Custom_Coadd_Sources.ipynb

```diff
@@ -0,0 +1,1248 @@
+    "\n",
+    "First, make a copy.\n",
+    "Otherwise, the second cell below will fail with the error message _\"Record data is not contiguous in memory.\"_"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "4376ec3a-724b-4128-b5b1-3c27fd51280e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sources = sources.copy(True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "4b315f6b-c81c-4984-85ed-505c7f31ecf2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "my_sources = sources.asAstropy()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6fb396ae-1187-41a2-be96-6838289fb139",
+   "metadata": {},
+   "source": [
+    "**Option:** print all of the column names of source measurements."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d5a7ba38-1c5a-4c0a-a14d-1c9357ff714d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# my_sources.colnames"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a547450d-be82-4f1d-b2c5-91e47ac06b35",
+   "metadata": {},
+   "source": [
+    "**Option:** explore which columns are flux measurements (not flux flags!) and which have been populated with nonzero flux values."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "17f06382-ed59-452e-9a5e-d2c61108841f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# for col in my_sources.colnames:\n",
+    "#     if ((col.find('Flux') >= 0) | (col.find('flux') >= 0)) & (col.find('flag') < 0):\n",
+    "#         tx = np.where(my_sources[col] > 0.0)[0]\n",
+    "#         print(len(tx), col)\n",
+    "#         del tx"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "17405e4e-87af-452c-b281-7413a9382746",
+   "metadata": {},
+   "source": [
+    "Convert instFlux to nanojanskies."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8aaa2e89-fd3c-48d8-905c-c3c6b1ecbf1f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "my_sources.add_column('i_CalibFlux_njy')\n",
+    "my_sources['i_CalibFlux_njy'] = np.zeros(len(my_sources), dtype='float')\n",
+    "my_deepCoadd_photoCalib = my_deepCoadd.getPhotoCalib()\n",
+    "for s in range(len(my_sources)):\n",
+    "    my_sources['i_CalibFlux_njy'][s] = \\\n",
+    "    my_deepCoadd_photoCalib.instFluxToNanojansky(my_sources['slot_CalibFlux_instFlux'][s])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "69b4fa27-8dca-49b5-93b1-5977aa11f2b5",
+   "metadata": {},
+   "source": [
+    "Plot the apparent i-band magnitude distribution of detected sources."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "aa6452ef-767a-426f-8759-6efd82e6bd5f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plt.figure(figsize=(5, 3))\n",
+    "plt.hist(-2.50 * np.log10(my_sources['i_CalibFlux_njy']) + 31.4, bins=20)\n",
```

Comment:
If we keep this around, it might be better to use astropy units to do the conversion, as that is more exact? (i.e., add units of nJy to the fluxes, then use `fluxes.to(u.ABmag)` to convert them).

Reply:
It wasn't kept, no.